### PR TITLE
#226 fix old slots mapping

### PR DESCRIPTION
**Pull request by SlakvichAleksandr**

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public AvailabilityProfile()
             CreateMap<AvailabilitySlot, AvailabilitySlotDto>()
                 .ForMember(dest => dest.Members, opt => opt.MapFrom(src => src.SlotMembers.Select(x => x.User)))
                 .ForMember(dest => dest.EmailTemplateSettings, opt => opt.MapFrom(src => src.EmailTemplates))
-                .ForMember(dest => dest.Schedule, opt => opt.MapFrom(s => s.SlotMembers.First().Schedule));
+                .ForMember(dest => dest.Schedule, opt => opt.MapFrom(s => s.SlotMembers.FirstOrDefault()!.Schedule));
             CreateMap<AvailabilitySlotDto, AvailabilitySlot>();
```
Comment (YuraRiabov) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs:17]: Wouldn't it still throw an error because of ! operator?

**Review by YuraRiabov — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs (hunk 1 above)
Comment (SlakvichAleksandr) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs:17]: I tested and didn't get any errors

Comment (YuraRiabov) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs:17]: Ok

**Review by YuraRiabov — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs (hunk 1 above)
Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/MappingProfiles/AvailabilityProfile.cs:17]: You cannot use ! with FirstOrDefault

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by SlakvichAleksandr** [comment]
If solution won't be optimal or inappropriate - here is a sql script for fixing data
[UpdateOldSlots.txt](https://github.com/BinaryStudioAcademy/bsa-2022-easymeets/files/9503475/UpdateOldSlots.txt)

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 2]
```diff
@@ -156,6 +156,11 @@ public async Task<AvailabilitySlotDto> GetAvailabilitySlotById(long id)
                 throw new KeyNotFoundException("Availability slot doesn't exist");
             }
 
+            if (!availabilitySlot.SlotMembers.Any())
+            {
+                availabilitySlot.SlotMembers.Add(new SlotMember { Schedule = new Schedule() });
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: This looks like a workaround than solutions

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 2 above)
Comment (SlakvichAleksandr) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Maybe. But in mapper profile we have rule that can broke endpoint if slot member collection will be empty. By this changes we don't affect anything and can return response

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: you need to find another way to fix it

Comment (SlakvichAleksandr) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: What about sql script that I attached upper ? Or it's also not a normal solution for this bug ?

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: You use hardcoded Ids this won't fix all developers' DBs

Comment (SlakvichAleksandr) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: As I understand, I need to solve this issue on our back-end side (without sql script, if we speak about all developers' DBs) ?

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: you can do it in the backend and in the DBs
Two cases are good and can be implemented

**Review by tatianahutii — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**